internal/workflow.go

```diff
@@ -1413,16 +1413,14 @@ func WithWorkflowTaskList(ctx Context, name string) Context {
 	return ctx1
 }
 
-// WithWorkflowTaskListMapper returns a copy of context and changes workflow tasklist with mapper function.
-func WithWorkflowTaskListMapper(ctx Context, mapper func(string) string) Context {
-	ctx1 := setWorkflowEnvOptionsIfNotExist(ctx)
-	var taskList string
-	if getWorkflowEnvOptions(ctx1).taskListName != nil {
-		taskList = *getWorkflowEnvOptions(ctx1).taskListName
+// GetWorkflowTaskList retrieves current workflow tasklist from context
+func GetWorkflowTaskList(ctx Context) *string {
+	wo := getWorkflowEnvOptions(ctx)
+	if wo == nil || wo.taskListName == nil {
+		return nil
 	}
-	newTaskList := mapper(taskList)
-	getWorkflowEnvOptions(ctx1).taskListName = &newTaskList
-	return ctx1
+	tl := *wo.taskListName // copy
+	return common.StringPtr(tl)
```

Comment:
```suggestion
	return &tl
```
might as well keep it simpler when we need that variable anyway